js/apps/admin-ui/src/organizations/LinkIdentityProviderModal.tsx

```diff
@@ -134,30 +153,42 @@ export const LinkIdentityProviderModal = ({
             isRequired
             isDisabled={!!identityProvider}
           />
-          <SelectControl
-            name={convertAttributeNameToForm("config.kc.org.domain")}
-            label={t("domain")}
-            controller={{ defaultValue: "" }}
-            options={[
-              { key: "", value: t("none") },
-              ...getValues("domains")!.map((d) => ({ key: d, value: d })),
-            ]}
-            menuAppendTo="parent"
+          <DefaultSwitchControl
+            name={convertAttributeNameToForm(
+              "config.kc.org.broker.redirect.mode.email-matches-any-org-domain",
+            )}
+            label={t("redirectWhenEmailMatchesAny")}
+            labelIcon={t("redirectWhenEmailMatchesAnyHelp")}
+            stringify
           />
+          {emailMatchAnyOrgDomain === "false" && (
+            <span>
```

Comment:
just use a fragment here instead of a extra dom element
```suggestion
            <>
```

Reply:
done. I also added a defaultValue for the watch.